perfkitbenchmarker/benchmark_spec.py

```diff
@@ -309,12 +314,15 @@ def PrepareVm(self, vm, firewall):
       firewall.AllowPort(vm, port)
     vm.AddMetadata(benchmark=self.benchmark_name)
     vm.WaitForBootCompletion()
-    vm.OnStartup()
     if FLAGS.scratch_disk_type == disk.LOCAL:
       vm.SetupLocalDisks()
     for disk_spec in vm.disk_specs:
       vm.CreateScratchDisk(disk_spec)
 
+    # This must come after Scratch Disk creation to support the
+    # Containerized VM case
+    vm.OnStartup()
```

Comment:
I think this will break RHEL for benchmarks that use disks - when they boot, you can't use sudo over ssh without using the '-t' option - we do some setup in the startup that gets around this. The Windows support I'm adding also expects OnStartup to be called immediately after boot before configuring the disks.